applications/nrf_desktop/src/hw_interface/motion_buttons.c

```diff
@@ -75,33 +98,63 @@ static enum dir key_to_dir(uint16_t key_id)
 	return dir;
 }
 
-static int16_t update_motion_dir(enum dir dir, uint32_t ts)
+static int16_t ts_diff_to_motion(int64_t diff, int32_t *reminder)
 {
-	uint32_t time_diff = ts - timestamp[dir];
+	int64_t res = diff * CONFIG_DESKTOP_MOTION_BUTTONS_MOTION_PER_SEC + *reminder;
 
-	if (time_diff > UCHAR_MAX) {
-		time_diff = UCHAR_MAX;
-	}
+	*reminder = res % sys_clock_hw_cycles_per_sec();
```

Comment:
Check the correctness in case of negative res values. See the following site

https://www.calculatorsoup.com/calculators/math/modulo-calculator.php
(a=-1320 b=1000 answer = 680)

Reply:
I think that it should work fine here. Please correct me if I'm wrong, but C standard section 6.5.5 seems to ensure that:
```
6) When integers are divided, the result of the / operator is the algebraic quotient with any
fractional part discarded.88) If the quotient a/b is representable, the expression
(a/b)*b + a%b shall equal a.

...

88) This is often called ‘‘truncation toward zero’’.
```

Reply:
(-1320 / 1000) * 1000 + mod_res = -1320
-2 * 1000 + mod_res = -1320
mod_res = 2000 - 1320 = 680

So you get the opposite sign of modulo result in comparison to the input value (-1320)
Do you see any issues with that?

Reply:
the `-1320/1000` would be `-1` and reminder would be -320`. Please double check the comment.

Reply:
I see thanks. I incorrectly assumed that the Python division operator works in a similar way as the C division operator.